import "galaxy.proto";
package baidu.galaxy;

option cc_generic_services = true;

message QueryRequest {
}

message QueryResponse {
    optional Status status = 1;
    optional AgentInfo agent = 2;
}

message RunPodRequest {
    optional string podid = 1;
    optional string jobid = 2;
    optional PodDescriptor pod = 3;
}

message RunPodResponse {
    optional Status status = 1;
}

message KillPodRequest {
    optional string podid = 1;
}

message KillPodResponse {
    optional Status status = 1;
}

service Agent {
    rpc Query(QueryRequest) returns (QueryResponse);
    rpc RunPod(RunPodRequest) returns (RunPodResponse);
    rpc KillPod(KillPodRequest) returns (KillPodResponse);
}
